utils/fork-tree/src/lib.rs

```diff
@@ -283,27 +251,16 @@ where
 		F: Fn(&H, &H) -> Result<bool, E>,
 		P: Fn(&V) -> bool,
 	{
-		// search for node starting from all roots
-		for root in self.roots.iter() {
-			let node = root.find_node_where(hash, number, is_descendent_of, predicate)?;
-
-			// found the node, early exit
-			if let FindOutcome::Found(node) = node {
-				return Ok(Some(node))
-			}
-		}
-
-		Ok(None)
-	}
-
-	/// Map fork tree into values of new types.
-	pub fn map<VT, F>(self, f: &mut F) -> ForkTree<H, N, VT>
-	where
-		F: FnMut(&H, &N, V) -> VT,
-	{
-		let roots = self.roots.into_iter().map(|root| root.map(f)).collect();
-
-		ForkTree { roots, best_finalized_number: self.best_finalized_number }
+		self.find_node_index_where(hash, number, is_descendent_of, predicate)
```

Comment:
I would have used `?` here to remove the `Result` and avoid one level of mapping.